### PR TITLE
chore: add bash script to handle update

### PR DESCRIPTION
**Pull request by gforsyth**
my bash is way out of date, but this is a simple script to handle
checking out the desired tag in the submodule, updating the version and
hash in the `__init__.py` file, and then running `gen_proto.sh`.

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+if [[ -z "$1" ]]; then
+    echo "Usage: bash update.sh version" 1>&2
```
Comment (EpsilonPrime) [inline update.sh:4]: I put replaceable arguments in angle brackets (`<version>`) but that's a personal style.  Up to you either way.

**Review by EpsilonPrime — APPROVED** [review]
The main thing to watch about bash is making sure that anything that calls out is properly escaped or quoted and that is being done here.

File: update.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    echo ""
+    echo "Example: bash update.sh v0.34.0" 1>&2
+    exit 1
+fi
+
+VERSION=$1
+
+pushd third_party/substrait/
```
Comment (mbrobbel) [inline update.sh:12]: This assumes that the working directory is the same as the script file location. You can make calling this script from any location work by adding:
```suggestion
DIR=$(cd $(dirname $0) && pwd)
pushd "${DIR}"/third_party/substrait/
```

File: update.sh (hunk 1 above)
Comment (gforsyth) [reply update.sh]: That's a good call. Added.

File: update.sh (hunk 2 above)
Comment (gforsyth) [reply update.sh]: Nice improvement, thanks @mbrobbel !

**Review by EpsilonPrime — APPROVED** [review]
lgtm

**Pull request merged**